tools/src/commands/GenerateBareApp.ts

```diff
@@ -90,9 +92,25 @@ async function createProjectDirectory({
   localTemplate: boolean;
 }) {
   if (localTemplate) {
+    // For local template, create an NPM package from the local template, then unpack it
+    // using the same transforms that @expo/cli uses
     const pathToBareTemplate = path.resolve(EXPO_DIR, 'templates', 'expo-template-bare-minimum');
-    const pathToWorkspace = path.resolve(workspaceDir, appName);
-    return fs.copy(pathToBareTemplate, pathToWorkspace, { recursive: true });
+    const pathToAppDirectory = path.resolve(workspaceDir, appName);
+    const templateVersion = require(path.join(pathToBareTemplate, 'package.json')).version;
+    await spawnAsync('npm', ['pack', '--pack-destination', workspaceDir], {
+      cwd: pathToBareTemplate,
+      stdio: 'ignore',
+    });
+    const tarFilePath = path.resolve(
+      workspaceDir,
+      `expo-template-bare-minimum-${templateVersion}.tgz`
+    );
+    fs.mkdirSync(pathToAppDirectory);
```

Comment:
```suggestion
    await fs.mkdirs(pathToAppDirectory);
```

Reply:
Interesting, why not `mkdirSync`?

Reply:
Synchronous calls for I/O operations is never a good idea. Think of the file system as the network, would you do network requests synchronously? Such synchronous call is also less efficient, most of the time it just wastes CPU resources on waiting for the output.
There are some cases where synchronous FS calls are necessary, but it's definitely not the case here.